modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -722,7 +737,10 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
 
   private def traitsToHints(traits: List[Trait]): List[Hint] = {
     val nonMetaTraits =
-      traits.filterNot(_.toShapeId().getNamespace() == "smithy4s.meta")
+      traits
+        .filterNot(_.toShapeId().getNamespace() == "smithy4s.meta")
+        // not sure about this
+        .filterNot(_.toShapeId().getNamespace() == "smithy.synthetic")
```

Comment:
I had to do it because `unfoldTrait` was throwing on a `None.get` when dealing with `smithy.synthetic.enum` (when I started calling `hints(shape)` with a 2.x enum shape). It might be possible to do it better.

Reply:
Mmm I'm not sure I understand. Can you share an example model that was leading it to throw ?

Reply:
Any 2.x enum with any hints. I can drop something later

Reply:
@Baccata question mark for when you have a sec

Reply:
Even this (already on the 0.17 branch) causes it:

```
enum SwitchState {
  ON
  OFF
}
```

Try [this commit](https://github.com/disneystreaming/smithy4s/commit/e2c090e741aa854e9381c3c7ad86ac61d4fd6445) and `example/smithy4sCodegen`.

Reply:
Okay, so turns out traits from the `smithy.synthetic` package do not have associated shapes in the model. I think the filter is okay, just change the comment to explain it.

Reply:
f274458cb44de0b126e9deeec60e5aba8cbff1b9